spec/support/model_helpers.rb

```diff
@@ -0,0 +1,26 @@
+# use like this:
+# it { is_expected.to have_validation :valid_date_presence,
+#                                     :on => :start_date,
+#                                     :options => [:allow_blank, :allow_nil]
+#                                     }
+RSpec::Matchers.define :have_validation do |validator, params|
+  validator_class = eval(validator.to_s.camelize + "Validator")
+
+  match do |model|
+    my_validators = model.class.validators.select { |v| v.class == validator_class and v.attributes.include? params[:on] }
```

Comment:
Best to use `&&` instead of `and` other than in cases of control flow (as in `redirect_to some_path and return`). Since the order of precedence is different between `&&` & `and`, you'll probably need parentheses around `v.class == validator_class`.